Makefile

```diff
@@ -1,27 +1,30 @@
-.PHONY: build clean redis-cli run shell stop up
+.PHONY: build clean clean-gke fixes gke help pip-compile pip-install-local redis-cli run shell stop test up
 
 
 help:
 	@echo "Welcome to the Telemetry Airflow\n"
 	@echo "The list of commands for local development:\n"
-	@echo "  build      		Builds the docker images for the docker-compose setup"
-	@echo "  clean      		Stops and removes all docker containers"
-	@echo "  pip-compile      	Compile dependencies from 'requirements.in' into 'requirements.txt'"
-	@echo "  pip-install-local	Install pip project requirements to your local environment"
-	@echo "  redis-cli  		Opens a Redis CLI"
-	@echo "  shell      		Opens a Bash shell"
-	@echo "  up         		Runs the whole stack, served under http://localhost:8080/"
-	@echo "  gke        		Create a sandbox gke cluster for testing"
-	@echo "  clean-gke  		Delete the sandbox gke cluster"
-	@echo "  stop       		Stops the docker containers"
+	@echo "  build              Builds the docker images for the docker-compose setup"
+	@echo "  clean              Stops and removes all docker containers"
+	@echo "  fixes              Applies Black and Ruff fixes to Python files"
+	@echo "  pip-compile        Compile dependencies from 'requirements.in' into 'requirements.txt'"
+	@echo "  pip-install-local  Install pip project requirements to your local environment"
+	@echo "  redis-cli          Opens a Redis CLI"
+	@echo "  shell              Opens a Bash shell"
+	@echo "  test               Runs pytest"
+	@echo "  up                 Runs the whole stack, served under http://localhost:8080/"
+	@echo "  gke                Create a sandbox gke cluster for testing"
+	@echo "  clean-gke          Delete the sandbox gke cluster"
+	@echo "  stop               Stops the docker containers"
 
 build:
 	docker-compose build
 
-pip-compile:
-	pip-compile --strip-extras --no-annotate
+pip-compile: pip-install-local
```

Comment:
This is creating a bit of chicken and egg situation where you need to install pip-tools via the output requirements, compile requirements.in and then reinstall updated dependencies via the new output requirements. IMO, pip-tools is a project prerequisite and it should be removed from the dev requirements.

A common pattern assumes the required package manager has to be installed outside of the application tooling. E.g. libraries and applications managed with poetry may have poetry listed in the README as a prerequisite, but poetry is never installed via pip within the repository. This should be covered by newcomers reading the README before contributing.

Reply:
I disagree:
- While Poetry is explicit that [it should be installed separately from the project](https://python-poetry.org/docs/#installation), `pip-tools` is equally explicit that [it should be installed in the same environment as the project](https://pip-tools.readthedocs.io/en/stable/#installation).
- I think it's important that we pin (or at least version constrain) all necessary tooling for consistency/reproducibility (I consider the fact that Poetry doesn't support version constraints on itself a deficiency). For example, CI currently always [installs the latest version of `pip-tools`](https://github.com/mozilla/telemetry-airflow/blob/1c463acbd56354a513f9a3073d421bfb71dc3f42/.circleci/config.yml#L107), and that could start failing at any time if a backwards incompatible version of `pip-tools` is released.
- Given the previous two points, IMO specifying `pip-tools` as a dev requirement pinned to a specific version is reasonable and prudent.

Reply:
`pip-tools` documentation mentions it needs to be installed in the project's virtual environment, but it does not indicate it needs to be managed within the project requirements. They cite `pip` as an analogous example; pip version is never managed as a project requirement. 

A quick google search lead me to this short [discussion](https://github.com/jazzband/pip-tools/issues/1679) where the accepted answer seems to be "both are fine". I agree that both options are acceptable and will work but I generally prefer the pattern of having package managers handled outside of the project dependencies. However that remains strictly a personal preference influenced by my past experience using poetry extensively.

> that could start failing at any time if a backwards incompatible version of pip-tools is released.

As far as I can recall, this happened once since I've been working on this repo and the fix was upgrading my local pip-tools version to regenerate requirements. The actual issue was in the requirements.txt file header changing due to a new `pip-tools` version. This should not happen anymore now that we use `--strip-extras --no-annotate` to generate requirements.

With all that said, I'm not opposed to the proposed changes.

Reply:
True, the `pip-tools` docs don't recommend managing it with the project requirements, but I think anything that materially impacts the project's virtual environment should be version controlled if possible (in both the source control and package management senses).  An example of why this is a concern is that installing the latest `pip-tools` would also install the `build`, `click`, `packaging`, `pyproject-hooks`, and `wheel` packages while ignoring the project's own constraints on any of those packages, which could have adverse effects.

And `pip` is a bit of a special case since it's distributed with Python itself.

Looking for other examples of how `pip-tools` is managed here at Mozilla, I see `pip-tools` is included in the requirements of a number of other projects like `bigquery-etl`, `gcp-ingestion-edge`, `pontoon`, `treeherder`, etc. (though IMO in some of those cases it should be moved to a separate dev requirements rather be included in the main requirements).